geotiepoints/simple_modis_interpolator.py

```diff
@@ -0,0 +1,255 @@
+
+    """
+    num_rows, num_cols = lon_array.shape
+    num_scans = int(num_rows / ROWS_PER_SCAN)
+    x_in, y_in, z_in = lonlat2xyz(lon_array, lat_array)
+
+    # Create an array of indexes that we want our result to have
+    x = np.arange(res_factor * num_cols, dtype=np.float32) * (1. / res_factor)
+    # 0.375 for 250m, 0.25 for 500m
+    y = np.arange(res_factor * ROWS_PER_SCAN, dtype=np.float32) * \
+        (1. / res_factor) - (res_factor * (1. / 16) + (1. / 8))
+    x, y = np.meshgrid(x, y)
+    coordinates = np.array([y, x])  # Used by map_coordinates, major optimization
+
+    new_x = np.empty((num_rows * res_factor, num_cols * res_factor), dtype=lon_array.dtype)
+    new_y = new_x.copy()
+    new_z = new_x.copy()
+    nav_arrays = [(x_in, new_x), (y_in, new_y), (z_in, new_z)]
+
+    # Interpolate each scan, one at a time, otherwise the math doesn't work well
+    for scan_idx in range(num_scans):
+        # Calculate indexes
+        j0 = ROWS_PER_SCAN * scan_idx
+        j1 = j0 + ROWS_PER_SCAN
+        k0 = ROWS_PER_SCAN * res_factor * scan_idx
+        k1 = k0 + ROWS_PER_SCAN * res_factor
+
+        for nav_array, result_array in nav_arrays:
+            # Use bilinear interpolation for all 250 meter pixels
+            map_coordinates(nav_array[j0:j1, :], coordinates, output=result_array[k0:k1, :], order=1, mode='nearest')
+
+            if res_factor == 4:
+                # Use linear extrapolation for the first two 250 meter pixels along track
+                m = (result_array[k0 + 5, :] - result_array[k0 + 2, :]) / (y[5, 0] - y[2, 0])
+                b = result_array[k0 + 5, :] - m * y[5, 0]
+                result_array[k0 + 0, :] = m * y[0, 0] + b
+                result_array[k0 + 1, :] = m * y[1, 0] + b
+
+                # Use linear extrapolation for the last  two 250 meter pixels along track
+                m = (result_array[k0 + 37, :] - result_array[k0 + 34, :]) / (y[37, 0] - y[34, 0])
+                b = result_array[k0 + 37, :] - m * y[37, 0]
+                result_array[k0 + 38, :] = m * y[38, 0] + b
+                result_array[k0 + 39, :] = m * y[39, 0] + b
+            else:
+                # 500m
+                # Use linear extrapolation for the first two 250 meter pixels along track
+                m = (result_array[k0 + 2, :] - result_array[k0 + 1, :]) / (y[2, 0] - y[1, 0])
+                b = result_array[k0 + 2, :] - m * y[2, 0]
+                result_array[k0 + 0, :] = m * y[0, 0] + b
+
+                # Use linear extrapolation for the last  two 250 meter pixels along track
+                m = (result_array[k0 + 18, :] - result_array[k0 + 17, :]) / (y[18, 0] - y[17, 0])
+                b = result_array[k0 + 18, :] - m * y[18, 0]
+                result_array[k0 + 19, :] = m * y[19, 0] + b
+
+    new_lons, new_lats = xyz2lonlat(new_x, new_y, new_z, low_lat_z=True)
+    return new_lons.astype(lon_array.dtype), new_lats.astype(lon_array.dtype)
+
+
+# def interpolate_geolocation(nav_array):
```

Comment:
Is this commented function necessary?

Reply:
No. I could have sworn I deleted this already. Removed it now anyway.